pyro/distributions/conjugate.py

```diff
@@ -77,12 +81,17 @@ def sample(self, sample_shape=()):
     def log_prob(self, value):
         if self._validate_args:
             self._validate_sample(value)
-        log_factorial_n = torch.lgamma(self.total_count + 1)
-        log_factorial_k = torch.lgamma(value + 1)
-        log_factorial_nmk = torch.lgamma(self.total_count - value + 1)
-        return (log_factorial_n - log_factorial_k - log_factorial_nmk +
-                _log_beta(value + self.concentration1, self.total_count - value + self.concentration0) -
-                _log_beta(self.concentration0, self.concentration1))
+
+        if self.approx_log_prob_tol > 0:
+            lbeta = functools.partial(log_beta_stirling, tol=self.approx_log_prob_tol)
+        else:
+            lbeta = log_beta
+
+        n = self.total_count
+        k = value
+        a = self.concentration1
+        b = self.concentration0
+        return lbeta(k + a, n - k + b) - lbeta(a, b) - lbeta(k + 1, n - k + 1) - n.log1p()
```

Comment:
I have refactored this math to use the `log_beta()` interface rather than calling `lgamma()` directly.